src/core/lib/security/authorization/grpc_authorization_engine.cc

```diff
@@ -20,27 +20,65 @@
 #include <map>
 #include <utility>
 
+#include <grpc/support/log.h>
+
+#include "src/core/lib/security/authorization/audit_logging.h"
+#include "src/core/lib/security/authorization/authorization_engine.h"
+
 namespace grpc_core {
 
+using experimental::AuditContext;
+using experimental::AuditLoggerRegistry;
+
+namespace {
+
+using Decision = AuthorizationEngine::Decision;
+
+bool ShouldLog(const Decision& decision,
+               const Rbac::AuditCondition& condition) {
+  return condition == Rbac::AuditCondition::kOnDenyAndAllow ||
+         (decision.type == Decision::Type::kAllow &&
+          condition == Rbac::AuditCondition::kOnAllow) ||
+         (decision.type == Decision::Type::kDeny &&
+          condition == Rbac::AuditCondition::kOnDeny);
+}
+
+}  // namespace
+
 GrpcAuthorizationEngine::GrpcAuthorizationEngine(Rbac policy)
-    : action_(policy.action) {
+    : name_(policy.name),
+      action_(policy.action),
+      audit_condition_(policy.audit_condition) {
   for (auto& sub_policy : policy.policies) {
     Policy policy;
     policy.name = sub_policy.first;
     policy.matcher = std::make_unique<PolicyAuthorizationMatcher>(
         std::move(sub_policy.second));
     policies_.push_back(std::move(policy));
   }
+  for (size_t i = 0; i < policy.logger_configs.size(); ++i) {
```

Comment:
`for (auto& logger_config : policy.logger_configs) {`

Reply:
Fixed.